src/main/java/io/github/linuxforhealth/core/expression/condition/CheckNotNull.java

```diff
@@ -31,7 +31,7 @@ public boolean test(Map<String, EvaluationResult> contextVariables) {
     EvaluationResult variable1 = ContextValueUtils.getVariableValuesFromVariableContextMap(var1,
         contextVariables, false, VariableUtils.isFuzzyMatch(var1));
     
-    return variable1 != null && !variable1.isEmpty();
+    return variable1 != null && !variable1.isEmpty() && !variable1.getValue().equals("null");
```

Comment:
Is this change strictly necessary?  Couldn't an HL7 value be literally the string `"null"` and then it should pass this check?

Reply:
Yes, it is necessary. The code will not work without it. I don't see why we wouldn't want to handle a string "null" as null anyways. It is null.

Reply:
Right... but in Java/Kotlin 

```java
"null" == null // false
```

not necessarily advocating to remove this, but just noting that this change removes the ability to differentiate between the two

Reply:
Yes, I understand that. I just don't see a case where we would not want to handle "null" as null.

Reply:
Yea, I think it's fine, but just thinking we should highlight this in the docs on conditions

Reply:
Where are the docs you are referring to located?

Reply:
I think its `TEMPLATING.md`

Reply:
What if someone's last name is Null? Or they live on Null Street? Or theres some sort of reference to a null hypothesis (Just making up unlikely scenarios...)

It's not likely but will be a royal pain to debug if we end up running into an issue.

Reply:
This doesn't process the actual message content. This processes the condition statement that we as developers may put on a field to determine whether or not that field should be sent through in the message or not.